File: CS/QualitativeScaleCustomSortOrderSample/Form1.cs
using DevExpress.XtraCharts;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace QualitativeScaleCustomSortOrderSample {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        #region #QualitativeScaleComparer
        private void OnLoad(object sender, EventArgs e) {
            XYDiagram diagram = chart.Diagram as XYDiagram;
            if (diagram == null) return;
            diagram.AxisX.QualitativeScaleComparer = new NumberComparer();
        }
        #endregion #QualitativeScaleComparer
    }
}

File: VB/QualitativeScaleCustomSortOrderSample/Form1.vb
Imports DevExpress.XtraCharts
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms

Namespace QualitativeScaleCustomSortOrderSample
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        #Region "#QualitativeScaleComparer"
        Private Overloads Sub OnLoad(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            Dim diagram As XYDiagram = TryCast(chart.Diagram, XYDiagram)
            If diagram Is Nothing Then
                Return
            End If
            diagram.AxisX.QualitativeScaleComparer = New NumberComparer()
        End Sub
        #End Region ' #QualitativeScaleComparer
    End Class
End Namespace
